client/blocks/app-banner/index.jsx

```diff
@@ -158,50 +148,59 @@ export class AppBanner extends Component {
 
 		return (
-			<div className={ classNames( 'app-banner-overlay' ) } ref={ this.preventNotificationsClose }>
-				<Card
-					className={ classNames( 'app-banner', 'is-compact', currentSection ) }
-					ref={ this.preventNotificationsClose }
-				>
-					<TrackComponentView
-						eventName="calypso_mobile_app_banner_impression"
-						eventProperties={ {
-							page: currentSection,
-						} }
-						statGroup="calypso_mobile_app_banner"
-						statName="impression"
-					/>
-					<div className="app-banner__circle is-top-left is-yellow" />
-					<div className="app-banner__circle is-top-right is-blue" />
-					<div className="app-banner__circle is-bottom-right is-red" />
-					<div className="app-banner__text-content">
-						<div className="app-banner__title">
-							<span> { title } </span>
-						</div>
-						<div className="app-banner__copy">
-							<span> { copy } </span>
-						</div>
-					</div>
-					<div className="app-banner__buttons">
-						<Button
-							primary
-							className="app-banner__open-button"
-							onClick={ this.openApp }
-							href={ this.getDeepLink() }
+			<ProvideExperimentData name="calypso_mobileweb_appbanner_frequency_20220128_v1">
+				{ ( isLoading, experimentAssignment ) => {
+					this.experimentIsControl = isLoading ? true : ! experimentAssignment?.variationName;
+
+					return (
+						<div
+							className={ classNames( 'app-banner-overlay' ) }
+							ref={ this.preventNotificationsClose }
 						>
-							{ translate( 'Open in app' ) }
-						</Button>
-						<Button className="app-banner__no-thanks-button" onClick={ this.dismiss }>
-							{ translate( 'No thanks' ) }
-						</Button>
-					</div>
-				</Card>
-			</div>
+							<Card
+								className={ classNames( 'app-banner', 'is-compact', currentSection ) }
+								ref={ this.preventNotificationsClose }
+							>
+								<TrackComponentView
+									eventName="calypso_mobile_app_banner_impression"
+									eventProperties={ {
+										page: currentSection,
+									} }
+									statGroup="calypso_mobile_app_banner"
+									statName="impression"
+								/>
+								<div className="app-banner__circle is-top-left is-yellow" />
+								<div className="app-banner__circle is-top-right is-blue" />
+								<div className="app-banner__circle is-bottom-right is-red" />
+								<div className="app-banner__text-content">
+									<div className="app-banner__title">
+										<span> { title } </span>
+									</div>
+									<div className="app-banner__copy">
+										<span> { copy } </span>
+									</div>
+								</div>
+								<div className="app-banner__buttons">
+									<Button
+										primary
+										className="app-banner__open-button"
+										onClick={ this.openApp }
+										href={ this.getDeepLink() }
+									>
+										{ translate( 'Open in app' ) }
+									</Button>
+									<Button className="app-banner__no-thanks-button" onClick={ this.dismiss }>
```

Comment:
We could pass now the experiment assignment via the dismiss handler:
```suggestion
									<Button className="app-banner__no-thanks-button" onClick={ this.dismiss.bind( null, ! experimentAssignment?.variationName ) }>
```
Updating the [dismiss method](https://github.com/Automattic/wp-calypso/blob/8f86bc0b963610f03adb7124f394fd86c8c98b44/client/blocks/app-banner/index.jsx#L109) to take `experimentIsControl` as the first argument would remove the need for `this.experimentIsControl` property.